### PR TITLE
[OpemVINO] Qwen3-ASR implement vlm like audio_encoder + text_embedding + language model export

### PR DESCRIPTION
**Pull request by as-suvorov**
# What does this PR do?

### Split Qwen3-ASR export into audio, embeddings, and language components

Refactor Qwen3-ASR export to produce openvino_audio_encoder_model.xml, openvino_text_embeddings_model.xml, and openvino_language_model.xml. Separating the language graph enables SDPA-to-PagedAttention transformation.

Reuse the Qwen3-Omni audio preprocessing and patcher, shared text-embedding export, and embedding-based language runtime.

Validated with Transformers 4.57.6, qwen-asr==0.0.6, and OpenVINO 2026.5.0 development build.

### Exported models

The export produces three model pairs, each consisting of an XML graph and BIN weights:

- `openvino_audio_encoder_model.xml`: encodes chunked mel features into audio embeddings.
- `openvino_text_embeddings_model.xml`: maps text token IDs to embeddings.
- `openvino_language_model.xml`: processes merged text/audio embeddings and produces vocabulary logits.

#### Audio encoder

| Direction | Name | Type | Shape | Description |
| --- | --- | --- | --- | --- |
| Input | `padded_feature` | float32 | `[num_chunks, 128, padded_chunk_frames]` | Mel features split into chunks of at most `2 * n_window` frames and padded across the batch. |
| Input | `padded_mask_after_cnn` | bool | `[num_chunks, padded_encoded_positions]` | Valid encoded positions in each chunk; `False` marks padding. |
| Input | `aftercnn_lens` | int64 | `[batch_size]` | Number of valid encoded positions per original recording. |
| Input | `cu_seqlens` | int32 | `[num_windows + 1]` | Cumulative attention-window boundaries in the compacted encoded sequence, starting at zero. |
| Output | `audio_features` | float32 | `[num_chunks, padded_encoded_positions, 1024]` | Padded audio embeddings projected into the language model's embedding space. |

The runtime prepares these inputs from processor outputs and removes padded output positions before replacing audio-placeholder embeddings in the text prompt. `num_chunks` counts chunks across all recordings in the batch.

The current audio patcher accepts `aftercnn_lens` and `cu_seqlens` but does not use them to control computation. Attention operates independently within each convolution chunk; larger attention-window grouping remains outside this PR.

#### Text embeddings

| Direction | Name | Type | Shape | Description |
| --- | --- | --- | --- | --- |
| Input | `input` | int64 | `[batch_size, current_tokens]` | Text token IDs; the exported port is named `input`. |
| Output | `inputs_embeds` | float32 | `[batch_size, current_tokens, 1024]` | Token embeddings before audio-placeholder positions are replaced with audio embeddings. |

#### Language model

| Direction | Name | Type | Shape | Description |
| --- | --- | --- | --- | --- |
| Input | `inputs_embeds` | float32 | `[batch_size, current_tokens, 1024]` | Merged text/audio embeddings during prefill; new token embeddings during decoding. |
| Input | `attention_mask` | int64 | `[batch_size, cached_tokens + current_tokens]` | Valid-token mask covering cached and current tokens. |
| Input | `position_ids` | int64 | `[3, batch_size, current_tokens]` | Three-axis rotary position IDs. The runtime supplies three axes, although the inspected IR declares all dimensions dynamic. |
| Input | `beam_idx` | int32 | `[batch_size]` | Cache-row indices for beam reordering; identity mapping for ordinary greedy decoding. |
| Output | `logits` | float32 | `[batch_size, current_tokens, 151936]` | Vocabulary scores for the tokens processed in the current call. |

### Usage

```sh
optimum-cli export openvino --model Qwen/Qwen3-ASR-0.6B \
  --task automatic-speech-recognition ./qwen3-asr-openvino
```

Batched inference uses AutoProcessor followed by generate, rather than the generic ASR pipeline. Audio is downloaded entirely in memory:

```python
import io
import librosa
import qwen_asr
import requests
import soundfile as sf
from transformers import AutoProcessor
from optimum.intel import OVModelForSpeechSeq2Seq

model_id = "Qwen/Qwen3-ASR-0.6B"
model = OVModelForSpeechSeq2Seq.from_pretrained(model_id, device="CPU")
processor = AutoProcessor.from_pretrained(model_id, fix_mistral_regex=True)
audio_urls = [
    "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen3-ASR-Repo/asr_en.wav",
    "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen3-ASR-Repo/asr_zh.wav",
]
sampling_rate = processor.feature_extractor.sampling_rate
audios = []
for url in audio_urls:
    response = requests.get(url, timeout=60)
    response.raise_for_status()
    audio, source_rate = sf.read(io.BytesIO(response.content), dtype="float32", always_2d=True)
    audios.append(librosa.resample(
        audio.mean(axis=1), orig_sr=source_rate, target_sr=sampling_rate
    ))

prompt = processor.apply_chat_template(
    [{"role": "system", "content": ""},
     {"role": "user", "content": [{"type": "audio"}]}],
    add_generation_prompt=True, tokenize=False,
)
inputs = processor(
    text=[prompt] * len(audios), audio=audios,
    sampling_rate=sampling_rate, padding=True, return_tensors="pt",
)
generated_ids = model.generate(
    **inputs, max_new_tokens=256, do_sample=False, eos_token_id=[151643, 151645]
)
print(processor.batch_decode(
    generated_ids[:, inputs.input_ids.shape[1]:], skip_special_tokens=True
))
```

### Accuracy for Qwen/Qwen3-ASR-0.6B

| Export | Similarity | WER | Word errors |
| --- | --- | --- | --- |
| Split (optimum) | 0.970588 | 2.94% | 14 / 476 |
| Legacy (optimum-main) | 0.962185 | 3.78% | 18 / 476 |

### LLM Bench results for Qwen/Qwen3-ASR-0.6B

| Export | 2nd tokens throughput (tokens/sec) | 
| --- | --- |
| Split (optimum) | 22.00  |
| Legacy (optimum-main) | 21.98  |

## Before submitting
- [NA] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [NA] Did you make sure to update the documentation with your changes?
- [X] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -10734,6 +10721,51 @@ def __exit__(self, exc_type, exc_value, traceback):
                 sparse_moe_block.forward = sparse_moe_block._orig_forward
 
 
+class Qwen3ASRLanguageModelPatcher(OVDecoderModelPatcher):
```
Comment (rkazants) [inline optimum/exporters/openvino/model_patcher.py:10740]: please add a comment - why do we need this patch? And link to original code line(s) that are patched.

File: tests/openvino/test_asr.py [hunk 2]
```diff
@@ -47,15 +52,42 @@ def _generate_audio_data(self):
...
+    def test_compare_to_transformers(self, model_arch, export_format):
         model_id = MODEL_NAMES[model_arch]
         set_seed(SEED)
 
         ref = self._get_pt_reference(model_arch)
 
-        ov_model = OVModelForSpeechSeq2Seq.from_pretrained(
-            model_id, export=True, trust_remote_code=True, ov_config=F32_CONFIG, device=OPENVINO_DEVICE
-        )
+        if export_format == "encoder_decoder":
```
Comment (rkazants) [inline tests/openvino/test_asr.py]: Please do not change this test suite. We want to have more generic and and less model specific.

**Review by rkazants — COMMENTED** [review]
please update test_cb_compatible.py

File: optimum/exporters/openvino/model_patcher.py (hunk 1 above)
Comment (as-suvorov) [reply optimum/exporters/openvino/model_patcher.py:10740]: Comment added

File: tests/openvino/test_asr.py (hunk 2 above)
Comment (as-suvorov) [reply tests/openvino/test_asr.py]: test removed

**Comment by as-suvorov** [comment]
> please update test_cb_compatible.py

qwen3-asr added to test_cb_compatible.py
